File: security-platform/agents/dotnet/SecurityPlatform.Agent/AgentFactory.cs
using System;
using System.IO;
using Microsoft.Extensions.Configuration;
using Microsoft.Extensions.Logging;

namespace SecurityPlatform.Agent
{
    /// <summary>
    /// Factory for creating Agent instances from configuration.
    /// </summary>
    public static class AgentFactory
    {
        /// <summary>
        /// Create agent from YAML configuration file.
        /// </summary>
        public static Agent FromYamlFile(string configPath, ILogger<Agent>? logger = null)
        {
            var configuration = new ConfigurationBuilder()
                .SetBasePath(Directory.GetCurrentDirectory())
                .AddYamlFile(configPath, optional: false)
                .Build();

            var config = new AgentConfig();
            configuration.Bind(config);
            return new Agent(config, logger);
        }

        /// <summary>
        /// Create agent from configuration object.
        /// </summary>
        public static Agent FromConfig(AgentConfig config, ILogger<Agent>? logger = null)
        {
            return new Agent(config, logger);
        }

        /// <summary>
        /// Create agent with default configuration from environment variables.
        /// </summary>
        public static Agent CreateDefault(ILogger<Agent>? logger = null)
        {
            var config = new AgentConfig
            {
                ControlPlaneUrl = Environment.GetEnvironmentVariable("SECURITY_PLATFORM_URL") 
                    ?? "https://api.securityplatform.com",
                AuthKey = Environment.GetEnvironmentVariable("SECURITY_PLATFORM_AUTH_KEY"),
                ServiceName = Environment.GetEnvironmentVariable("SERVICE_NAME") ?? "dotnet-service",
                Environment = Environment.GetEnvironmentVariable("ENVIRONMENT") ?? "production"
            };
            return new Agent(config, logger);
        }
    }
}

File: security-platform/agents/java/src/main/java/com/securityplatform/agent/AgentFactory.java
package com.securityplatform.agent;

import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.io.File;
import java.io.InputStream;

/**
 * Factory for creating Agent instances from configuration.
 */
public class AgentFactory {
    private static final Logger logger = LoggerFactory.getLogger(AgentFactory.class);
    private static final ObjectMapper yamlMapper = new ObjectMapper(new YAMLFactory());
    private static final ObjectMapper jsonMapper = new ObjectMapper();

    /**
     * Create agent from YAML configuration file.
     */
    public static Agent fromYamlFile(String configPath) {
        try {
            AgentConfig config = yamlMapper.readValue(new File(configPath), AgentConfig.class);
            return new Agent(config);
        } catch (Exception e) {
            logger.error("Failed to load config from file: " + configPath, e);
            throw new RuntimeException("Failed to load agent configuration", e);
        }
    }

    /**
     * Create agent from YAML configuration stream.
     */
    public static Agent fromYamlStream(InputStream configStream) {
        try {
            AgentConfig config = yamlMapper.readValue(configStream, AgentConfig.class);
            return new Agent(config);
        } catch (Exception e) {
            logger.error("Failed to load config from stream", e);
            throw new RuntimeException("Failed to load agent configuration", e);
        }
    }

    /**
     * Create agent from JSON configuration file.
     */
    public static Agent fromJsonFile(String configPath) {
        try {
            AgentConfig config = jsonMapper.readValue(new File(configPath), AgentConfig.class);
            return new Agent(config);
        } catch (Exception e) {
            logger.error("Failed to load config from file: " + configPath, e);
            throw new RuntimeException("Failed to load agent configuration", e);
        }
    }

    /**
     * Create agent from configuration object.
     */
    public static Agent fromConfig(AgentConfig config) {
        return new Agent(config);
    }

    /**
     * Create agent with default configuration.
     */
    public static Agent createDefault() {
        AgentConfig config = new AgentConfig();
        config.setControlPlaneUrl(System.getenv().getOrDefault("SECURITY_PLATFORM_URL", "https://api.securityplatform.com"));
        config.setAuthKey(System.getenv("SECURITY_PLATFORM_AUTH_KEY"));
        config.setServiceName(System.getenv().getOrDefault("SERVICE_NAME", "java-service"));
        config.setEnvironment(System.getenv().getOrDefault("ENVIRONMENT", "production"));
        return new Agent(config);
    }
}
